### PR TITLE
Feature Enhancement - Github Issue: Clarify usage for key reference 

### PR DESCRIPTION
**Pull request by khushal08**
"Feature Enhancement - Github Issue: Clarify usage for key reference vs actual name of the resource"
Details Here - https://github.com/Azure/terraform-azurerm-avm-ptn-virtualwan/issues/9

### CODE REVIEW

**Review by jtracey93 — APPROVED** [review]
LGTM

**Review by jaredfholgate — CHANGES_REQUESTED** [review]
The code changes look good, but it looks like the docs have not been updated?

Your repo is missing all the standard AVM workflows that would check for this. They need to be added. You can use grept or I can help you.

It might be worth expanding on the docs in here also: https://github.com/Azure/terraform-azurerm-avm-ptn-virtualwan/blob/main/_header.md

Perhaps explain the usage in a bit more detail for people unfamiliar with virtual wan and this module?

File: firewall.tf [hunk 1]
```diff
@@ -1,10 +1,10 @@
 resource "azurerm_firewall" "fw" {
   for_each = var.firewalls
 
-  location            = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].location
-  resource_group_name = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].resource_group_name
+  location            = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_map_key].location
+  resource_group_name = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_map_key].resource_group_name
   virtual_hub {
-    virtual_hub_id = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].id
+    virtual_hub_id = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_map_key].id
```
Comment (matt-FFFFFF) [inline firewall.tf:7]: Can we call this _resource_id instead of just _id?

File: p2s-vpn-gateway.tf [hunk 2]
```diff
@@ -18,9 +18,9 @@ resource "azurerm_point_to_site_vpn_gateway" "p2s_gateway" {
   for_each = local.p2s_gateways != null && length(local.p2s_gateways) > 0 ? local.p2s_gateways : {}
 
   name                        = each.value.name
-  location                    = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].location
-  resource_group_name         = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].resource_group_name
-  virtual_hub_id              = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].id
+  location                    = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_map_key].location
+  resource_group_name         = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_map_key].resource_group_name
+  virtual_hub_id              = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_map_key].id
```
Comment (matt-FFFFFF) [inline p2s-vpn-gateway.tf:23]: Same here.. use _resource_id

**Review by matt-FFFFFF — COMMENTED** [review]
Some recommended changes

**Comment by jaredfholgate** [comment]
@khushal08 I think this PR can be closed now as I updated the naming for all the key vs name variables I could find in #23. So this one is already taken care of.

**Pull request closed without merging**